addon_service/abstract/authorized_account/models.py

```diff
@@ -207,12 +206,17 @@ def api_base_url(self) -> str:
     @api_base_url.setter
     def api_base_url(self, value: str):
         self._api_base_url = (
-            "" if value == self.external_service.api_base_url else value
+            "" if (value == self.external_service.api_base_url or not value) else value
         )
 
     @property
     def imp_cls(self) -> type[AddonImp]:
-        return self.external_service.addon_imp.imp_cls
+        if isinstance(self.external_service, ExternalService):
```

Comment:
if we do stay on this path, i'd assume that `ExternalService` should use multi-table inheritance as well  -- then `AuthorizedAccount.external_service` would be a ForeignKey instead of an abstract method, and this `imp_cls` method could remain unchanged